site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1912,8 +1911,12 @@ public function getPostsForThread($current_user, $thread_id){
           $thread_id = $announcement_id;
         }
       }
-
-      $this->course_db->query("SELECT * FROM posts WHERE thread_id=? AND deleted = false ORDER BY timestamp ASC", array($thread_id));
+      if($option == 'alpha'){
+        $this->course_db->query("SELECT posts.*, users.user_lastname FROM posts INNER JOIN users ON posts.author_user_id=users.user_id WHERE thread_id=? ORDER BY user_lastname;", array($thread_id));
```

Comment:
Even if `post` is `Anonymous` it is still being sorted according to author's `last_name`.

Reply:
Since the visual of the student remains anonymous and only Full Access Graders and Instructors have access to this option I believe that this behavior is fine for now. I am going to make a note of this though and we can possibly change this in the future.